components/post_view/post_time/index.ts

```diff
@@ -4,12 +4,14 @@
 import {connect} from 'react-redux';
 import {getCurrentRelativeTeamUrl} from 'mattermost-redux/selectors/entities/teams';
 
-import PostTime from './post_time.jsx';
+import {GlobalState} from 'types/store';
 
-function mapStateToProps(state) {
+import PostTime from './post_time';
+
+function mapStateToProps(state: GlobalState) {
     return {
         teamUrl: getCurrentRelativeTeamUrl(state),
     };
 }
 
-export default connect(mapStateToProps)(PostTime);
+export default connect(mapStateToProps)(PostTime as any);
```

Comment:
Why do we need this casting?

Reply:
There was a ts error that I didn't know how to fix: Error:(17, 41) TS2345: Argument of type 'typeof PostTime' is not assignable to parameter of type 'ComponentType<never>'.
  Type 'typeof PostTime' is not assignable to type 'ComponentClass<never, any>'.
    Types of property 'defaultProps' are incompatible.
      Type '{ eventTime: number; location: string; }' is not assignable to type 'undefined'.

Reply:
@jakubnovak998 Your component expects no props, so your connect has an error as it infers that `mapStateToProps` should return an empty object. Try typing the defaultProps object and creating an interface 
```
type defaultProps = {
    eventTime: number,
    location: \\ whatever the location type is
}

export default class PostTime extends React.PureComponent<Props & defaultProps> {
   \\ ...
}
```

Reply:
@jakubnovak998 Can you post the error you're getting without casting as `any`?